### ISSUE TITLE
Bug: ast-spec should allow AwaitExpression and NewExpression in LeftHandSideExpression

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6239]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&showAST=es&code=MYewdgzgLgBApgDwIYFsAOAbOMC8MDeAvjEhCWAJ4DcAUDQBRIDuSAlrIqpnAJQB0aAE4g0tRi3YwwcJvGTos-ISNo1EaEINj4YhKkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkeAhpgObp4VTolHoAebANOBqQSeTMmh4AcgFdUqAMIALdNADWZSmJnohAXxB6gA

### Repro Code

```typescript
const example = {} as any;

(await example).prop;
(await new example).prop;
```

### Expected Result

In the type system: the `object` property in the `TSESTree.MemberExpression` type is a `TSESTree.LeftHandSideExpression`.

In the playground's tree: the `object` property on the first `MemberExpression` is an `AwaitExpression`. The second `MemberExpression`'s `object` property is a `NewExpression`.

`TSESTree.LeftHandSideExpression` should therefore allow `TSESTree.AwaitExpression` and `TSESTree.NewExpression`.

### Actual Result

The `TSESTree.LeftHandSideExpression` union does not include `TSESTree.AwaitExpression` or `TSESTree.NewExpression`.

### Additional Info

Originally spotted in https://github.com/typescript-eslint/typescript-eslint/pull/6028/files#r1032600861

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/ast-spec`     | `5.46.1` |
| `@typescript-eslint/types`     | `5.46.1` |
| `@typescript-eslint/typescript-estree` | `5.46.1` |

**Comment by bradzacher** [comment]
Same as https://github.com/typescript-eslint/typescript-eslint/issues/6192

It's supposed to be `Expression` not `LeftHandSideExpression`

**Comment by JoshuaKGoldberg** [comment]
Ah great!
